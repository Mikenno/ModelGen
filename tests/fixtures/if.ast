Module
- Call
-- Name print
-- String "begin"
- If
-- Name true
-- Call
--- Name print
--- String "true"
- If
-- Name true
-- Call
--- Name print
--- String "true"
- Call
-- Name print
-- String "end"

- Call
-- Name print
-- String "begin"
- If
-- Name true
-- Call
--- Name print
--- String "true"
-- Call
--- Name print
--- String "false"
- Call
-- Name print
-- String "end"

- Call
-- Name print
-- String "begin"
- If
-- Name true
-- Block
--- Call
---- Name print
---- Integer 1
--- Call
---- Name print
---- Integer 2
-- Block
--- Call
---- Name print
---- Integer 3
--- Call
---- Name print
---- Integer 4
- Call
-- Name print
-- String "end"

- Assign
-- Name x
-- Integer 1
- Assign
-- Name y
-- Integer 1

- Call
-- Name print
-- String "begin"
- If
-- BinOpEq
--- Name x
--- Name y
-- Call
--- Name print
--- String "true"
-- Call
--- Name print
--- String "false"
- Call
-- Name print
-- String "end"

- Call
-- Name print
-- String "begin"
- If
-- BinOpEq
--- Name x
--- Name y
-- Block
--- Call
---- Name print
---- String "true"
--- If
---- BinOpEq
----- Name x
----- Name y
---- Call
----- Name print
----- String "true true"
--- Call
---- Name print
---- String "true"
-- Block
--- Call
---- Name print
---- String "false"
--- If
---- BinOpEq
----- Name x
----- Name y
---- Call
----- Name print
----- String "false true"
---- Call
----- Name print
----- String "false false"
--- Call
---- Name print
---- String "false"
- Call
-- Name print
-- String "end"

- For
-- Name i
-- Call
--- Name range
--- Integer 4
-- Block
--- Call
---- Name print
---- Name i
--- If
---- BinOpGreater
----- Name i
----- Integer 1
---- Block
----- Call
------ Name print
------ Name i
------ Integer 1
----- If
------ BinOpGreater
------- Name i
------- Integer 2
------ Call
------- Name print
------- Name i
------- Integer 1
------- Integer 1
---- Block
----- Call
------ Name print
------ Name i
------ Integer 0
----- If
------ BinOpGreater
------- Name i
------- Integer 0
------ Call
------- Name print
------- Name i
------- Integer 0
------- Integer 1
------ Call
------- Name print
------- Name i
------- Integer 0
------- Integer 0

- Call
-- Name print
-- Integer 1
- If
-- Integer 2
- Call
-- Name print
-- Integer 3

- If
-- Integer 1
-- If
--- Integer 2
--- If
---- Integer 3
---- NoOperation
---- Call
----- Name print
----- String "else 1"
-- If
--- Integer 4
--- If
---- Integer 5
---- If
----- Integer 6
----- If
------ Integer 7
----- If
------ Integer 8
------ NoOperation
------ Call
------- Name print
------- String "else 2"
--- Call
---- Name print
---- String "else 3"