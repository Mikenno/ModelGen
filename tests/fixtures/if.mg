
print("begin")
if true
	print("true")
if true print("true")
print("end")

print("begin")
if true print("true")
else print("false")
print("end")

print("begin")
if true print(1) print(2)
else print(3) print(4)
print("end")

x = 1
y = 1

print("begin")
if x == y
	print("true")
else
	print("false")
print("end")

print("begin")
if x == y
	print("true")
	if x == y
		print("true true")
	print("true")
else
	print("false")
	if x == y
		print("false true")
	else
		print("false false")
	print("false")
print("end")

for i in range(4)
	print(i)
	if i > 1
		print(i, 1)
		if i > 2
			print(i, 1, 1)
	else
		print(i, 0)
		if i > 0
			print(i, 0, 1)
		else
			print(i, 0, 0)

print(1)
if 2
else
print(3)

if 1
	if 2
	else if 3
	else
		print("else 1")
else if 4
	if 5
	else if 6
		if 7
		else
	else if 8
	else
		print("else 2")
else
	print("else 3")
