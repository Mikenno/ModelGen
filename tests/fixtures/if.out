begin
true
true
end
begin
true
end
begin
1
2
end
begin
true
end
begin
true
true true
true
end
0
0 0
0 0 0
1
1 0
1 0 1
2
2 1
3
3 1
3 1 1
1
3
